src/Microsoft.OData.Core/RawValueWriter.cs

```diff
@@ -173,8 +173,8 @@ private void InitializeTextWriter()
             {
                 nonDisposingStream = MessageStreamWrapper.CreateNonDisposingStream(this.stream);
             }
-
-            this.textWriter = new StreamWriter(nonDisposingStream, this.encoding);
+            
+            this.textWriter = new StreamWriter(nonDisposingStream, this.encoding) { NewLine = this.settings.MultipartNewLine };

```

Comment:
>(nonDisposingStream, this.encoding) { NewLine = this.settings.MultipartNewLine }; [](start = 46, length = 81)

RawValueWriter only writes "Primitive, Enum" value, both of them are single line value. (correct me @mikepizzo).

Reply:
This is how Raw Text writer is coming into the picture when using ODataBatchWriter:


`abstract class ODataBatchWriter` has an abstract method `abstract void WriteStartChangesetImplementation ();`

Which is overwritten in the `internal sealed class ODataMultipartMixedBatchWriter : ODataBatchWriter`

ODataMultipartMixedBatchWriter derives from ODataRawOutputContext whic has a TextWriter property

`class ODataMultipartMixedBatchOutputContext : ODataRawOutputContext`
` internal TextWriter TextWriter { get    { return this.rawValueWriter.TextWriter; }

Which is initialized in line 259
`   this.rawValueWriter = new RawValueWriter( … `

The implementation of for example WriteStartChangesetImplementation uses the static ( ☹ ) `ODataMultipartMixedBatchWriterUtils.WriteStartBoundary`

Line 194 of ODataMultipartMixedBatchWriterUtils
`       ODataMultipartMixedBatchWriterUtils.WriteStartBoundary(this.RawOutputContext.TextWriter, ` …

and this calls for example:  ` writer.WriteLine("--{0}", boundary);`

Reply:
If we don't think it would break clients, the ideal fix would be to have an internal lineEnding property on rawValueWriter, which defaults to the current platform-dependent line ending, and set that to /r/n when called from ODataMultipartMixedBatchWriterUtils. What we are currently writing is not compliant to the http multipart/mixed specification.

---
In reply to: [463729331](https://github.com/OData/odata.net/pull/1844#discussion_r463729331) [](ancestors = 463729331)

Reply:
This is roughly the direction I was taking. 
But I did not want to have a property on RawValueWriter but pass it in the constructor. Instead of adding a new parameter I found it appropriate to pass the flag in as part of the settings.